### PR TITLE
Update configuration docs

### PR DESCRIPTION
**Pull request by dnephin**
Fix section names, add internal links,

### CODE REVIEW

File: docs/yaml.md [hunk 1]
```diff
@@ -9,35 +9,37 @@ Each section adds file to the root file system
...
 which should contain a `bzImage` (for `amd64` architecture, others may vary) and a file
 called `kernel.tar` which is a tarball that is unpacked into the root, which should usually
 contain a kernel modules directory. See [`kernel/`](../kernel/) for source code. `cmdline`
 specifies the kernel command line options if required.
 
 ## `init`
 
-This section currently just lists images that is used for the `init` system and are unpacked directly
+The `init` section is a list of images that are used for the `init` system and are unpacked directly
 into the root filesystem. This should bring up `containerd`, start the system and daemon containers,
```
Comment (dnephin) [inline docs/yaml.md]: *should* or *must* include `containerd` ? What happens if it doesn't ?

File: docs/yaml.md [hunk 2]
```diff
@@ -9,35 +9,37 @@ Each section adds file to the root file system
...
 rather than bundled into the `init` container.
 
 ## `onboot`
 
-These containers are run to completion sequentially, using `runc` before anything else is started.
-They can be used to configure one shot settings. For details of the config for each container, see
-below.
+The `onboot` section is a list of images. These images are run before any other
+images. They are run sequentially and each must exit before the next one is run.
+These images can be used to configure one shot settings. See [Image
```
Comment (dnephin) [inline docs/yaml.md]: What is a one shot setting?

File: docs/yaml.md (hunk 1 above)
Comment (justincormack) [inline docs/yaml.md]: you can do something totally different with linxukit, but its up to you to make it work...

File: docs/yaml.md (hunk 2 above)
Comment (justincormack) [inline docs/yaml.md]: not very clear, but just something that gets set once.

**Comment by justincormack** [comment]
Still needs work but lets merge these fixes anyway. This should move to `moby/tool`.

**Pull request merged**